docs: add pr template (#89)
shopware/SwagMigrationAssistant: .github/PULL_REQUEST_TEMPLATE.md

> ### 1. Why is this change necessary?

Reviewer: ```suggestion
### 1. Why is this change necessary?

// Please link to the relevant issues, if they exists, or explain the solved issue here. e.g.
// closes https://github.com/shopware/shopware/issues/{ISSUE_NUMBER}
```

> ### 2. What does this change do, exactly?
> 

Reviewer: ```suggestion
### 2. What does this change do, exactly?

// Shortly explain how your change solves the issue
// If the issue has more than one obvious solution,
// shortly explain your reasoning for your chosen approach
```

### 3. Describe each step to reproduce the issue or behaviour.

> ### 4. Please link to the relevant issues (if any).

Reviewer: Can be removed if covered by 1. https://github.com/shopware/SwagMigrationAssistant/pull/89/files#r2526092590

### 5. Checklist

> 

Reviewer: ```suggestion

- [ ] I have created the PR in draft status and only open it when it's ready for review
- [ ] If the change is large: I have thought about splitting it up into smaller PRs
```

> - [ ] I have written tests and verified that they fail without my change

Reviewer: ```suggestion
- [ ] I have written tests and if it's a bug fix verified that they fail without my change or that new code is covered by tests
```

- [ ] I have updated developer-facing release notes if this change affects external developers
- [ ] I have written or adjusted the documentation according to my changes

> - [ ] This change has comments for package types, values, functions, and non-obvious lines of code

Reviewer: package and types are annotations and the latter should be enforced by the linter / PHPStan. So I would rewrite this point to something like this

```suggestion
- [ ] I added the correct package annotation to each `src` file (not strictly necessary for tests)
- [ ] This change has code comments where appropriate, especially for non-obvious lines of code
```

> - [ ] Ensure this pull request follows repository conventions (e.g. conventional commits)

Reviewer: Maybe also move it to the top next to the point about opening as draft
```suggestion
- [ ] Ensure the pull request title as well as first commit follows conventional commits
```
Also: do we even need to mention this? I guess the PR title is already checked? But the first commit? 🤔

> - [ ] Ensure this pull request follows repository conventions (e.g. conventional commits)

Reviewer: I would add one more point:

```
- [ ] I have thought about well defined extension points and marked everything else as `@internal` / `@private`
```
